### DISCUSSION TITLE
Warning "This transaction could steal your funds in the future" when using Durable & Offline Transaction Signing using Nonces

### DISCUSSION THREAD

**Discussion by yeemKantapong** [phantom/docs#205] [Help & Support] [answered]
### Summary

![SCR-20240329-khns](https://github.com/phantom/docs/assets/111940141/bea5fdb7-6498-4a21-8c81-038eb70763d9)

### Question
What is a way so that Phantom does not show this warning to user? In terms of like, how can I make Phantom knows that I'm not able to steal funds in the future?

### Description
From playing around with the transaction instructions. I can see that this warning "This transaction could steal your funds in the future" only display if `nonceAdvance` instruction is added in the transaction. I do understand that technically, this transaction can be sent sometime in the future since this is a durable nonce transaction. However, I don't think that I could steal user funds, apart from all the funds in the transaction (i.e. transfer xx SOL).

Also, it would also make sense that Phantom only shows this warning if i use `signTransaction` function since this means that transaction can be sent in the future. For my case, I use `sendTransaction`, which means that user sign and send the transaction at once, therefore I'm not able to steal funds in the future. So maybe Phantom should not show this warning?

### Steps to Reproduce

1. I (as a backend) construct a transaction with these
   - `instruction:nonceAdvance` - assuming nonce account is created before
   - `instructions:transfer` - i.e. 0.001 SOL from my wallet to user wallet
   - Set fee payer to user
   - Set blockhash to nonce value

2. Build and Serialize the transaction, then sign with my keypair
3. User then deserialize, sign and send the transaction - this is where the warning shows up

### Phantom Version

24.4.1

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment, marked as answer]
@yeemKantapong Over the past few days there were some issues with our transaction simulation. Would you mind confirming that you are still seeing this behavior?

**Reply by yeemKantapong** [reply]
You are correct it seems like on devnet, the warning is now gone. 

But a new warning shows up "There was a problem getting your token balance for this transaction. It may fail", however, I think its another issue that is not about durable offline transaction or maybe it is?

<img width="362" alt="SCR-20240402-kccf" src="https://github.com/phantom/docs/assets/111940141/fbaf40a1-e793-4f06-bbf3-e84fe7d4e85e">

**Reply by adamdelphantom** [reply]
@yeemKantapong Few questions:

 Is this on devnet or main net? 

Are you still seeing this issue? 

Is it only with this token? If so could you provide the token address, please?

**Reply by drez** [reply]
Still there!
Devnet, havent tested mainnet.
I am doing a SOL transaction with nonce.
Happened on this: 2ZJLDPn1dH47Cwgzs3jTYf5NdwkJGwFcCeq2rYnZczYNPYMTYqpHF4Ub12f8S4t4MNB2GHZ5YQVSoWe5ZRyC1P6S

**Comment by MexicanSparker** [comment]
Hello. I am building a POC on Solana Durable Nonces but I receive this message when signing the transaction by the user. It doesn't happen with a normal tx, as soon as I add the nonceAdvance instruction I get this message.
![Phantom_Wallet_and__engineering___Supersafe_Labs_-_Discord](https://github.com/user-attachments/assets/d52b8288-0c4a-4e44-b347-7f8176d8a5b6)
The transaction is the simplest possible: A system transfer of 0.05 SOL from my connected wallet to another address. This is on devnet, I haven't tried on mainnet just yet.
My POC is running from my local dev environment
Here is the submitted transaction: 3YWRWpsYddePrF6Y7KnGzvjTsrscUFkhT4VsHMpzKPHj5oRaCRCysVUKNtHijhoh9zmqN8Tjz2MQnPazgzFC5wov
